app/src/main/java/com/github/se/eventradar/ui/login/SignUpScreen.kt

```diff
@@ -0,0 +1,467 @@
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.constraintlayout.compose.ConstraintLayout
+import coil.compose.rememberImagePainter
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract
+import com.github.se.eventradar.R
+import com.github.se.eventradar.ui.navigation.NavigationActions
+import com.github.se.eventradar.ui.navigation.Route
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+import com.google.firebase.firestore.firestore
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.text.SimpleDateFormat
+
+private const val TAG = "SignUpScreen"
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SignUpScreen(navigationActions: NavigationActions) {
+
+    val openErrorDialog = remember { mutableStateOf(false) }
+
+    val context = LocalContext.current
+    val selectedImageUri = rememberSaveable { mutableStateOf<Uri?>(null) }
+    val username = rememberSaveable { mutableStateOf("") }
+    val name = rememberSaveable { mutableStateOf("") }
+    val surname = rememberSaveable { mutableStateOf("") }
+    val phoneNumber = rememberSaveable { mutableStateOf("") }
+    val birthDate = rememberSaveable { mutableStateOf("") }
+    val selectedCountryCode = rememberSaveable { mutableStateOf("+1") } // Default country code
+    // List of country codes. You can replace this with your own list.
+    val countryCodes = listOf("+1", "+33", "+41", "+44", "+91", "+61", "+81")
+
+    val launcher = rememberLauncherForActivityResult(
+        contract = FirebaseAuthUIActivityResultContract(),
+        onResult = { result ->
+            if (result.resultCode == Activity.RESULT_OK) {
+                val user = Firebase.auth.currentUser
+                val userValues = hashMapOf(
+                    "Name" to name.value,
+                    "Surname" to surname.value,
+                    "Phone Number" to phoneNumber.value,
+                    "Birth Date" to birthDate.value,
+                    "Email" to user?.email,
+                    "Profile Picture" to selectedImageUri.value.toString()
+                )
```

Comment:
It will be yes. I have separated this PR into 2. This PR now should only focus on the UI design as well as reproducing the behaviour of the login button from Nikhen. The other will be focused around the storing of User Data (both locally, so with a `User.kt` class, and remotely through the Firestore Database. I took note of this for the other PR!